packages/relay/src/lib/clients/sdkClient.ts

```diff
@@ -498,75 +503,125 @@ export class SDKClient {
     }
   }
 
+  /**
+   * Executes transaction -> handle errors -> capture burned HBAR in metrics and HBAR rate limit class.
+   * @param transaction
+   * @param callerName
+   * @param interactingEntity
+   * @param requestId
+   * @returns
+   */
   async executeTransaction(
     transaction: Transaction,
     callerName: string,
     interactingEntity: string,
     requestId: string,
-  ): Promise<TransactionResponse> {
+  ): Promise<TransactionResponse | TransactionResponse[]> {
```

Comment:
Please, define another method - `executeAllTransactions` which returns `TransactionResponse[]`. The `executeTransaction` method already has too many responsibilities and adding this change makes it even harder to maintain. 

Also, defining a separate method for executing multiple transactions can help us with type safety and avoid casting later inside `submitEthereumTransaction`:

Example usage:
```typescript
  async submitEthereumTransaction(
    transactionBuffer: Uint8Array,
    callerName: string,
    requestId: string,
  ): Promise<{ txResponses: TransactionResponse[]; fileId: FileId | null }> {
    ...
    if (transaction.constructor.name === FileAppendTransaction.name) {
      return {
        fileId,
        txResponses: await this.executeAllTransactions(
          ethereumTransaction,
          callerName,
          interactingEntity,
          requestId,
        ),
      };
    } else {
      return {
        fileId,
        txResponses: [
          await this.executeTransaction(
            ethereumTransaction,
            callerName,
            interactingEntity,
            requestId,
          ),
        ],
      };
    }
  }
```

Reply:
Thanks for the input! Yeah I see and agree that splitting executeTransaction up and put executeAll() logic into a new method would make the code more maintainable. Pushing in updates soon!